src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/QueryExecutor.java

```diff
@@ -121,24 +121,69 @@ public Uni<AsyncResultSet> executeWrite(SimpleStatement statement) {
   }
 
   /**
-   * Execute schema change query with bound statement.
+   * Execute schema change query with bound statement for create ddl statements.
    *
    * @param boundStatement - Bound statement with query and parameters. The table name used in the
    *     query must have keyspace prefixed.
    * @return AsyncResultSet
    */
-  public Uni<AsyncResultSet> executeSchemaChange(SimpleStatement boundStatement) {
+  public Uni<AsyncResultSet> executeCreateSchemaChange(SimpleStatement boundStatement) {
+    return executeSchemaChange(boundStatement, "create");
+  }
+
+  /**
+   * Execute schema change query with bound statement for drop ddl statements.
+   *
+   * @param boundStatement - Bound statement with query and parameters. The table name used in the
+   *     query must have keyspace prefixed.
+   * @return AsyncResultSet
+   */
+  public Uni<AsyncResultSet> executeDropSchemaChange(SimpleStatement boundStatement) {
+    return executeSchemaChange(boundStatement, "drop");
+  }
+
+  /**
+   * Execute schema change query with bound statement for drop ddl statements.
+   *
+   * @param boundStatement - Bound statement with query and parameters. The table name used in the
+   *     query must have keyspace prefixed.
+   * @return AsyncResultSet
+   */
+  public Uni<AsyncResultSet> executeTruncateSchemaChange(SimpleStatement boundStatement) {
+    return executeSchemaChange(boundStatement, "truncate");
+  }
+
+  private Uni<AsyncResultSet> executeSchemaChange(SimpleStatement boundStatement, String profile) {
     return Uni.createFrom()
         .completionStage(
             cqlSessionCache
                 .getSession()
                 .executeAsync(
                     boundStatement
-                        .setExecutionProfileName("ddl")
+                        .setExecutionProfileName(profile)
                         .setIdempotent(true)
                         .setSerialConsistencyLevel(
                             operationsConfig.queriesConfig().consistency().schemaChanges())))
         .onFailure(DriverTimeoutException.class)
+        .recoverWithUni(
+            throwable -> {
+              logger.error("Timeout executing schema change query");
```

Comment:
Shouldn't we include information on what kind of query was being executed?

Reply:
Added the query

Reply:
Thanks! Looks good, approved.